src/main/cpp/src/get_json_object.hpp

```diff
@@ -19,20 +19,270 @@
 #include <cudf/strings/string_view.hpp>
 #include <cudf/strings/strings_column_view.hpp>
 
+#include <thrust/optional.h>
+
 #include <memory>
 
 namespace spark_rapids_jni {
 
+namespace detail {
+
+/**
+ * path instruction type
+ */
+enum class path_instruction_type { SUBSCRIPT, WILDCARD, KEY, INDEX, NAMED };
+
 /**
- * Extracts json object from a json string based on json path specified, and
- * returns json string of the extracted json object. It will return null if the
- * input json string is invalid.
+ * path instruction
  */
+struct path_instruction {
+  CUDF_HOST_DEVICE inline path_instruction(path_instruction_type _type) : type(_type) {}
+
+  path_instruction_type type;
+
+  // used when type is named type
+  cudf::string_view name{"", 0};
+
+  // used when type is index
+  long index{-1};
+};
+
+class JsonPathParser {
```

Comment:
Would it be better to just pass down the parsed version from Spark? This feels like it is providing very little value over the Spark version.